core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -261,4 +267,55 @@ private Connector createConnector(ConnectionConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  private ConnectionConfig resolveConfig(ConnectionConfig config) {
```

Comment:
Could we encapsulate this into ConnectionConfig? Something like `connectionConfig.getKey()` and then do this:

```
return connectors.computeIfAbsent(config.getKey(), k -> createConnector(config));
```

Reply:
No, because `ConnectionConfig` is in the public API package and `ConnectionKey` is in the internal package. We want `ConnectionKey` to only be used internally, and to change as needed.